### PR TITLE
Fix/cell fac course

### PR DESCRIPTION
**Pull request by hredestig**
Tons of changes here, best to click on each commit and view them separately as purpose is indicated that way (saves me making 10+ PRs..)

- sort various minor bugs (divide by zero and lingering usages of add_demand )
- move duplicated features with cobrapy out of cameo (pfba, find_esssential_x)
- remove `cameo.stuff`, will it be missed?

### CODE REVIEW

File: cameo/strain_design/heuristic/evolutionary/objective_functions.py [hunk 1]
```diff
@@ -208,15 +208,13 @@ def __call__(self, model, solution, targets):
...
             product_flux = round(solution.fluxes[self.product], config.ndecimals)
             substrate_flux = round(sum(abs(solution.fluxes[s]) for s in self.substrates), config.ndecimals)
-        try:
+        if substrate_flux > config.non_zero_flux_threshold:
             return (biomass_flux * product_flux) / substrate_flux
-
-        except ZeroDivisionError:
-            return 0.0
+        else:
+            return 0.
```
Comment (joaocardoso) [inline cameo/strain_design/heuristic/evolutionary/objective_functions.py]: For now let's leave it like this, but it should return `self.worst_fitness` later. But for that we need to make the evaluation function aware of the direction of the optimization.

File: cameo/strain_design/deterministic/linear_programming.py [hunk 2]
```diff
@@ -130,11 +131,11 @@ def _reduce_to_nullspace(self, reactions):
     def _build_problem(self, essential_reactions, use_nullspace_simplification):
         logger.debug("Starting to formulate OptKnock problem")
 
-        self.essential_reactions = find_essential_reactions(self._model) + self._model.exchanges
+        self.essential_reactions = find_essential_reactions(self._model).union(self._model.exchanges)
```
Comment (joaocardoso) [inline cameo/strain_design/deterministic/linear_programming.py]: Does the model still has the `essential_reactions`? If yes, then it should call `self._model.essential_reactions()`

File: cameo/strain_design/heuristic/evolutionary/objective_functions.py (hunk 1 above)
Comment (hredestig) [inline cameo/strain_design/heuristic/evolutionary/objective_functions.py]: Alright, this PR was about getting rid of the ZeroDivError so yep, let's handle the improvement in a different pr

File: cameo/strain_design/deterministic/linear_programming.py (hunk 2 above)
Comment (hredestig) [inline cameo/strain_design/deterministic/linear_programming.py]: No, `essential_reactions` is not a method of cobra `Model` didn't really see the need for it there..

**Comment by hredestig** [comment]
Sure thing, plotting_old is removed

**Pull request merged**